lib/logstash/outputs/elasticsearch/common.rb

```diff
@@ -115,6 +117,64 @@ def setup_hosts
       end
     end
 
+    def hosts_default?(hosts)
+      # NOTE: would be nice if pipeline allowed us a clean way to detect a config default :
+      hosts.is_a?(Array) && hosts.size == 1 && hosts.first.equal?(CommonConfigs::DEFAULT_HOST)
+    end
+    private :hosts_default?
+
+    def fill_hosts_from_cloud_id
+      return unless @cloud_id
+
+      if @hosts && !hosts_default?(@hosts)
+        raise LogStash::ConfigurationError, 'Both cloud_id and hosts specified, please only use one of those.'
+      end
+      @hosts = parse_host_uri_from_cloud_id(@cloud_id)
+    end
+
+    def fill_user_password_from_cloud_auth
+      return unless @cloud_auth
+
+      if @user || @password
+        raise LogStash::ConfigurationError, 'Both cloud_auth and user/password specified, please only use one.'
+      end
+      @user, @password = parse_user_password_from_cloud_auth(@cloud_auth)
+      params['user'], params['password'] = @user, @password
+    end
+
+    def parse_host_uri_from_cloud_id(cloud_id)
+      begin # might not be available on older LS
+        require 'logstash/util/cloud_setting_id'
+      rescue LoadError
+        raise LogStash::ConfigurationError, 'The cloud.id setting is not supported by LogStash, please set hosts instead.'
```

Comment:
Nits: `cloud.id` should read `cloud_id` to reflect configuration property name
          Capitalization of "LogStash" should read Logstash
          Maybe state that the setting is not supported by this version of Logstash, as later versions do support it.